src/hex_tarball.erl

```diff
@@ -134,6 +134,42 @@ unpack(Tarball, Output) ->
             {error, {tarball, Reason}}
     end.
 
+%% @doc
+%% Unpacks a documentation tarball.
+%%
+%% Examples:
+%%
+%% ```
+%% > hex_tarball:unpack_docs(Tarball, memory).
+%% {ok,#{checksum => <<...>>,
+%%       contents => [{"src/foo.erl",<<"-module(foo).">>}],
+%%       metadata => #{<<"name">> => <<"foo">>, ...}}}
+%%
+%% > hex_tarball:unpack_docs(Tarball, "path/to/unpack").
+%% {ok,#{checksum => <<...>>,
+%%       metadata => #{<<"name">> => <<"foo">>, ...}}}
+%% '''
+-spec unpack_docs(tarball(), memory) ->
+                {ok, #{checksum => checksum(), metadata => metadata(), contents => contents()}} |
```

Comment:
Docs tarball has neither checksum nor metadata. I think the spec should be:

```erlang
-spec unpack_docs(tarball(), memory) -> {ok, contents()} | {error, term()};
                 (tarball(), filename()) -> ok | {error, term()}.
```

Thought I'm not sure about using type `tarball()` here. What I mean is, does it make sense to differentiate between _package_ tarball and _docs_ tarball in types? The former is a tar of checksum, metadata.config, contents.tar.gz; the latter is an _compressed_ tar of actual docs files. So maybe we should have a separate `docs_tarball()` type? 

I'm ok with re-using the same type, they're both binaries under the hood but curious what others think. cc @ericmj @ferd @tsloughter

Reply:
structurally I bet the types will be identical (they're both tarballs), so what you have here is pretty much as good as it gets for analysis. The only distinction you could make is add aliases to signify intent (i.e. `-type docs_tarball() :: tarball()` which would at the very least let you make a distinction similar to meters and feet even if they'd all be numbers.

I do prefer stricter types in such a case because the whole thing is closer to self-documenting, but it's got no analysis benefits (I don't think Dialyzer is fancy enough to track it)